### PR TITLE
Fix aggregation metrics with zero tensors

### LINKED ISSUE #1063: Aggregation metrics ignore zeros tensors
## 🐛 Bug

When feeding `CatMetric`, `MinMetric` or `MaxMetric` with zeros tensor, the metric isn't updated.

### To Reproduce

1. Create a new `CatMetric` instance and feed it with zeros tensor.
2. `compute()` returns empty list.

#### Code sample

```python
import torch
import torchmetrics

metric = torchmetrics.CatMetric()
metric(torch.zeros((5, )))
print(metric.compute()) # outputs []
```

### Expected behavior

Zeros tensor should update metric's state.

### Environment

```
torch==1.11.0
torchmetrics==0.9.0
```
Ubuntu 20.04.4

### Additional context
The problem is caused by these lines:
https://github.com/PyTorchLightning/metrics/blob/567fa3405d5a60ba0a6052e57afe5a5d34c376a7/torchmetrics/aggregation.py#L139
https://github.com/PyTorchLightning/metrics/blob/567fa3405d5a60ba0a6052e57afe5a5d34c376a7/torchmetrics/aggregation.py#L188
https://github.com/PyTorchLightning/metrics/blob/567fa3405d5a60ba0a6052e57afe5a5d34c376a7/torchmetrics/aggregation.py#L280

`any(value.flatten())` evaluates to `False` on zeros tensors.

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #1063
Change `if any(value)` to `if value.numel() != 0` to check if metric is empty.
Also adds the `full_state_update` to aggregation metrics as it was missing.

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — APPROVED** [review]

File: torchmetrics/aggregation.py [hunk 1]
```diff
@@ -136,7 +139,7 @@ def update(self, value: Union[float, Tensor]) -> None:  # type: ignore
                 dimensions will be flattened
         """
         value = self._cast_and_nan_check_input(value)
-        if any(value.flatten()):  # make sure tensor not empty
+        if value.numel() != 0:  # make sure tensor not empty
```
Comment (justusschock) [inline torchmetrics/aggregation.py:142]:
```suggestion
        if value.numel():  # make sure tensor not empty
```

File: torchmetrics/aggregation.py [hunk 2]
```diff
@@ -185,7 +190,7 @@ def update(self, value: Union[float, Tensor]) -> None:  # type: ignore
...
```
Comment (justusschock) [inline torchmetrics/aggregation.py:193]:
```suggestion
        if value.numel():  # make sure tensor not empty
```

File: torchmetrics/aggregation.py [hunk 3]
```diff
@@ -234,7 +239,8 @@ def update(self, value: Union[float, Tensor]) -> None:  # type: ignore
                 dimensions will be flattened
         """
         value = self._cast_and_nan_check_input(value)
-        self.value += value.sum()
+        if value.numel() != 0:
```
Comment (justusschock) [inline torchmetrics/aggregation.py:242]:
```suggestion
        if value.numel():
```

**Review by justusschock — APPROVED** [review]
